libmamba/src/core/package_handling.cpp

```diff
@@ -354,14 +356,53 @@ namespace mamba
         LOG_INFO << "Extracting " << file << " to " << destination;
         extraction_guard g(destination);
 
+        struct archive* a;
+        int r;
+
+        a = archive_read_new();
+        archive_read_support_format_tar(a);
+        archive_read_support_format_zip(a);
+        archive_read_support_filter_all(a);
+
+        auto lock = LockFile(file);
+        r = archive_read_open_filename(a, file.string().c_str(), 10240);
+
+        if (r != ARCHIVE_OK)
+        {
+            LOG_ERROR << "Error opening archive: " << archive_error_string(a);
+            throw std::runtime_error(file.string() + " : Could not open archive for reading.");
+        }
+
+        stream_extract_archive(a, destination);
+
+        archive_read_close(a);
+        archive_read_free(a);
+    }
+
+    namespace
+    {
+        struct conda_extract_context
+        {
+            conda_extract_context(struct archive* source)
+                : source(source)
+                , buffer(ZSTD_DStreamOutSize())
+            {
+            }
+
+            struct archive* source;
+            std::vector<char> buffer;
+        };
+    }
+
+    void stream_extract_archive(archive* a, const fs::u8path& destination)
+    {
         auto prev_path = fs::current_path();
         if (!fs::exists(destination))
         {
             fs::create_directories(destination);
         }
         fs::current_path(destination);
 
-        struct archive* a;
         struct archive* ext;
         struct archive_entry* entry;
         int flags;
```

Comment:
Same remark here, when possible we should avoid variable definition without immediatly initializing them.